pkg/executor/build.go

```diff
@@ -228,3 +231,24 @@ func resolveOnBuild(stage *config.KanikoStage, config *v1.Config) error {
 	config.OnBuild = nil
 	return nil
 }
+
+// reviewConfig makes sure the value of CMD is correct after building the stage
+// If ENTRYPOINT was set in this stage but CMD wasn't, then CMD should be cleared out
+// See Issue #346 for more info
+func reviewConfig(stage config.KanikoStage, config *v1.Config) error {
```

Comment:
This function does not encounter an error flow and always returns nil.
There is no need to return an error.

Reply:
done